torch_choice/model/formula_parser.py

```diff
@@ -0,0 +1,58 @@
+from torch_choice.data.choice_dataset import ChoiceDataset
+from typing import Tuple, Dict
+
+
+def parse_formula(formula: str, data: ChoiceDataset) -> Tuple[Dict[str, int], Dict[str, int]]:
+    """Generates the coef_variation_dict and the num_samples_dict for Conditional/Nested logit models from a R-like
+    formula and a ChoiceDataset.
+    The parser reads variations of coefficients for different observables from the formula, then the parser retrieves
+    the number of parameters for each coefficient (i.e., the dimension of each observable) from the ChoiceDataset.
+
+    Args:
+        formula (str): a string representing the utility formula.
+            The formula consists of '(variable_name|variation)'s separated by '+', for example:
+            "(var1|item) + (var2|user) + (var3|constant)"
```

Comment:
i think delete spaces here since they don't count; i also think the parentheses are better added to bemb for notational consistency between the packages, but that is for later down the line.